processor/src/tests/stdlib/crypto/sha256.rs

```diff
@@ -0,0 +1,80 @@
+use crate::{execute, Felt, FieldElement, ProgramInputs, Script, MIN_STACK_DEPTH};
+use sha2::{Digest, Sha256};
+use vm_core::utils::IntoBytes;
+
+#[test]
+fn sha256_2_to_1_hash() {
+    let script = compile(
+        "
+        use.std::crypto::hashes::sha256
+
+        begin
+            exec.sha256::hash
+        end",
+    );
+
+    // prepare random input byte array
+    let i_digest_0: [u8; 32] = rand_utils::rand_array::<Felt, 4>().into_bytes();
+    let i_digest_1: [u8; 32] = rand_utils::rand_array::<Felt, 4>().into_bytes();
+
+    // two digests concatenated to form input to sha256 2-to-1 hash function
+    let mut i_digest = [0u8; 64];
+    i_digest[..32].copy_from_slice(&i_digest_0);
+    i_digest[32..].copy_from_slice(&i_digest_1);
+
+    // allocate space on stack so that bytes can be converted to sha256 words
+    let mut i_words = [0u64; MIN_STACK_DEPTH];
+
+    // convert each of four consecutive big endian bytes (of input) to sha256 words
+    for i in 0..MIN_STACK_DEPTH {
+        i_words[i] = from_be_bytes_to_words(&i_digest[i * 4..(i + 1) * 4]) as u64;
+    }
+    i_words.reverse();
+
+    let mut hasher = Sha256::new();
+    hasher.update(&i_digest);
+    let digest = hasher.finalize();
+
+    // prepare digest in desired sha256 word form so that assertion writing becomes easier
+    let mut digest_words = [0u64; MIN_STACK_DEPTH >> 1];
+    // convert each of four consecutive big endian bytes (of digest) to sha256 words
+    for i in 0..(MIN_STACK_DEPTH >> 1) {
+        digest_words[i] = from_be_bytes_to_words(&digest[i * 4..(i + 1) * 4]) as u64;
+    }
+
+    // finally execute miden program on VM
+    let inputs = ProgramInputs::new(&i_words, &[], Vec::new()).unwrap();
+    let trace = execute(&script, &inputs).unwrap();
+    let last_state = trace.last_stack_state();
```

Comment:
You might be able to use the new `build_test` macro from [here](https://github.com/maticnetwork/miden/blob/next/processor/src/tests/mod.rs#L197) to simplify this a bit.

Reply:
Thanks, changed here https://github.com/maticnetwork/miden/pull/123/commits/d46ae2e7180ab789803282511ecd0ac07b0ed777